File: IPL_WEEK_4/D/Main.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		int n = in.nextInt();
		char diag = '?';
		char ot = '?';
		for(int i = 0; i < n; i++){
			String s = in.nextLine();
			for(int j = 0; j < n; j++){
				char c = s.charAt(j);
				if(i == j || n-i-1 == j){
					if(diag == '?'){
						diag = c;
					}
					else if(diag != c){
						System.out.println("NO");
						return;
					}
				}
				else {
					if(ot == '?'){
						ot = c;
					}
					else if(ot != c){
						System.out.println("NO i");
						return;
					}
			
				}
			}

		}
		System.out.println("YES");
	}
}

File: IPL_WEEK_4/D/main.cpp
#include<iostream>
#include<string>
using namespace std;
int main() {
	int n; 
	cin >> n;
	char x = '?';
	char o = '?';
	for(int i = 0; i< n; i++){
		string s;
		cin >> s;
		for(int j = 0; j <n; j++){
			char c = s[j];
			if(i==j||i == n-j-1){
				if(x == '?'){
					x = c;
				}
				else if(x != c) {
					cout <<"NO"<<endl;
					return 0;
				}
			}
			else {
				if(o == '?'){
					o = c;
				}
				else if(o != c){
					cout<<"NO"<<endl;
					return 0;
				}
			}
		}
	}
	if(x == o){
		cout<<"NO"<<endl;
		return 0;
	}
	cout<<"YES"<<endl;
	return 0;
}
